pkg/ebpf/c/common/buffer.h

```diff
@@ -10,6 +10,7 @@
 // PROTOTYPES
 
 statfunc buf_t *get_buf(int);
+statfunc data_filter_lpm_key_t *get_data_filter_buf(int);
```

Comment:
`get_data_filter_buf` will return a `data_filter_lpm_key_t` struct, which internally holds a `str` field (renamed from `pathname`). We plan to add other types of fields inside `data_filter_lpm_key_t` soon, so I think it's better to use a more generic name like `get_data_filter_buf`. Today make more sense to call `get_string_data_filter_buf` but I think soon will be better `get_data_filter_buf`. WDYT?

Reply:
Yeah... today, since we only have string filters, the data filter is very tightly coupled with the string filter, so `get_string_data_filter_buf ` make sense... but on other hand, we soon will add other field type filters, so maybe we should let it generic as `get_data_filter_buf`? WDYT?

Reply:
But prefix filter will only be relevant to string filters, won't it?
What other field types may fit LPM trie?

Reply:
After some refactoring, I think it makes sense to have separate functions. I changed them to `get_string_data_filter_buf` and `get_string_data_filter_lpm_buf`. I'm also considering renaming some variables and maps, such as `data_filter_lpm_key_t` (to `str_data_filter_lpm_key_t`). WDYT?